src/Components/HomePage.js

```diff
@@ -0,0 +1,157 @@
+import { MdSettings } from 'react-icons/md';
+import { FaMicrophone } from 'react-icons/fa';
+import { BsArrowRightCircle } from 'react-icons/bs';
+import { getAllCoins, getCoinMarkets, searchByRank } from '../Redux/CoinAPI';
+import cryptoPix from './images/crypto-pix.jpg';
+
+const HomePage = () => {
+  const getList = useSelector((state) => state.cryptoCoins);
+
+  const coins = getList.coin;
+  const { currency } = getList;
+
+  const dispatch = useDispatch();
+
+  let cur = 'USD';
+
+  const onSelect = (e) => {
+    cur = e.target.value;
+    dispatch(getAllCoins(cur));
+    const parent = e.target.parentElement.children[1];
+    parent.children[0].value = '';
+  };
+
+  useEffect(() => {
+    dispatch(getAllCoins(cur));
+  }, []);
+
+  let rank = '';
+
+  const onFillValue = (e) => {
+    rank = e.target.value;
+  };
+
+  const onSearch = (e) => {
+    e.preventDefault();
+
+    if (rank) {
+      dispatch(searchByRank(rank));
+    } else if (!rank) {
+      dispatch(getAllCoins(cur));
+    }
+  };
+
+  const onClear = (e) => {
+    dispatch(getAllCoins(cur));
+    e.target.parentElement.firstElementChild.value = '';
+  };
+
+  const onBackspace = (e) => {
+    if (e.key === 'Backspace') {
+      dispatch(getAllCoins(cur));
+    }
+  };
+
+  const handleClick = (e) => {
+    dispatch(getCoinMarkets(e.target.id));
+  };
+
+  return (
+    <div className="pg-container">
+      <nav>
+        <p className="heading">Top Cryptos</p>
+        <div className="nav-icon">
+          <FaMicrophone style={{ color: 'white', marginRight: '2rem' }} />
+          <MdSettings style={{ color: 'white' }} />
+        </div>
+      </nav>
+      <div className="pg-head">
+        <div className="crypto-pix">
+          <img className="image" src={cryptoPix} alt="currency" />
+        </div>
+        <div className="head-desr">
+          <h1 className="crypto-x">CRYPTO EXCHANGE</h1>
+        </div>
+
+      </div>
+
+      <div className="coin-heading">
+        <div>
+          <h4>Price Per Coin</h4>
+        </div>
+
+        <div className="search-row">
+
+          <select
+            className="currency-name"
+            name="currency"
+            type="text"
+            onChange={(e) => onSelect(e)}
+          >
+            <option value="USD">Choose Currency</option>
+            <option value="USD">USD - US Dollars</option>
+            <option value="EUR">EUR - EURO</option>
+            <option value="GBP">GBP - British Pounds</option>
+            <option value="CAD">CAD - Canadian Dollars</option>
+            <option value="AUD">AUD - Australian Dollars</option>
+          </select>
+
+          <form onSubmit={(e) => onSearch(e)}>
+            <input
+              name="search"
+              type="number"
+              className="search-field"
+              placeholder="Search Top Rank"
+              onChange={(e) => onFillValue(e)}
+              onKeyDown={(e) => onBackspace(e)}
+              max="50"
+            />
```

Comment:
- [ ] I can not type inside the input, this is because the input does not have any `value` attribute as it is supposed to be a controlled input, kindly follow this [link](https://reactjs.org/docs/forms.html) to get a clearer view about controlled and uncontrolled input

Reply:
Hi @VICTOR-CHUKWU,

Thank you for your review. I designed the input field to accept only numbers to search the top-ranking cryptos. Adding value attributes will not allow users to enter the number of their choice for the app to filter through.  

kindly try entering a number on the search field to see the result. You can as well explain to me how I can implement the value attribute to the filter field that will also allow users to enter any number of their choice on the field.

I am still learning and will welcome any recommendation that will improve my application.

I look forward to hearing from you.

Reply:
nice work, it works now :+1: